<!DOCTYPE html>
<html lang="en-us">
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <meta name="robots" content="noodp" />
        <meta http-equiv="X-UA-Compatible" content="IE=edge, chrome=1">
        <title>Genome Analysis | Op1um</title>
        <meta name="Description" content="A blog about computers and other stuff."><meta property="og:title" content="Genome Analysis" />
<meta property="og:description" content="I have long hesitated to order an analysis of my genome. I found it interesting to learn more about myself through genome. But on the other hand I am against the way in which cheap genome analysis companies try to use such personal data. In this article I explain how I had my genome analyzed, I give some advice to protect your personal data as much as possible, and how I proceeded to manually do a much deeper analysis of the raw genome data." />
<meta property="og:type" content="article" />
<meta property="og:url" content="https://op1um.blog/genome-analysis/" />
<meta property="article:published_time" content="2020-03-26T13:23:09+01:00" />
<meta property="article:modified_time" content="2020-03-26T13:23:09+01:00" />
<meta name="twitter:card" content="summary"/>
<meta name="twitter:title" content="Genome Analysis"/>
<meta name="twitter:description" content="I have long hesitated to order an analysis of my genome. I found it interesting to learn more about myself through genome. But on the other hand I am against the way in which cheap genome analysis companies try to use such personal data. In this article I explain how I had my genome analyzed, I give some advice to protect your personal data as much as possible, and how I proceeded to manually do a much deeper analysis of the raw genome data."/>
<meta name="theme-color" content="#ffffff">
<meta name="msapplication-TileColor" content="#da532c">
<link rel="canonical" href="https://op1um.blog/genome-analysis/" />
<link rel="shortcut icon" type="image/x-icon" href="/favicon.ico" />
<link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
<link rel="manifest" href="/site.webmanifest">
<link rel="mask-icon" href="/safari-pinned-tab.svg" color="#5bbad5"><link rel="next" href="https://op1um.blog/osint-find-people-using-tinder/" /><link rel="stylesheet" href="/lib/fontawesome-free/all.min.css"><link rel="stylesheet" href="/lib/animate/animate.min.css"><link rel="stylesheet" href="/css/style.min.css"><script type="application/ld+json">
    {
        "@context": "http://schema.org",
        "@type": "BlogPosting",
        "headline": "Genome Analysis",
        "inLanguage": "en-us",
        "mainEntityOfPage": {
            "@type": "WebPage",
            "@id": "https:\/\/op1um.blog\/genome-analysis\/"
        },"image": {
                "@type": "ImageObject",
                "url": "https:\/\/op1um.blog\/cover.png",
                "width":  800 ,
                "height":  600 
            },"genre": "posts","keywords": "open-source, genome, dna, python","wordcount":  737 ,
        "url": "https:\/\/op1um.blog\/genome-analysis\/","datePublished": "2020-03-26","dateModified": "2020-03-26","publisher": {
                "@type": "Organization",
                "name": "Samuel Cohen",
                "logo": {
                "@type": "ImageObject",
                "url": "https:\/\/op1um.blog\/logo.png",
                "width":  127 ,
                "height":  40 
                }
            },"author": {
                "@type": "Person",
                "name": "Op1um"
            },"description": ""
    }
    </script></head>
    <body><script>
            if (!window.localStorage || !window.localStorage.getItem('theme')) {window.isDark = 'dark' === 'dark';} else {
                window.isDark = (window.localStorage && window.localStorage.getItem('theme')) === 'dark';
            }
            window.isDark && document.body.classList.add('dark-theme');
        </script><div class="wrapper"><header class="desktop" id="header-desktop">
    <div class="header-wrapper">
        <div class="header-title">
            <a href="/">Op1um</a>
        </div>
        <div class="menu"><a class="menu-item" href="/posts/" rel="noopener noreffer">Posts</a><a class="menu-item" href="/tags/" rel="noopener noreffer">Tags</a><a class="menu-item" href="/categories/" rel="noopener noreffer">Categories</a><a class="menu-item" href="/about/" rel="noopener noreffer">About</a><span class="menu-item">|</span>
            <a href="javascript:void(0);" class="menu-item theme-switch" title="Switch Theme">
                <i class="fas fa-adjust fa-fw"></i>
            </a></div>
    </div>
</header><header class="mobile" id="header-mobile">
    <div class="header-wrapper">
        <div class="header-container">
            <div class="header-title">
                <a href="/">Op1um</a>
            </div>
            <div class="menu-toggle" id="menu-toggle-mobile">
                <span></span><span></span><span></span>
            </div>
        </div>
        <div class="menu" id="menu-mobile"><a class="menu-item" href="/posts/" title="" rel="noopener noreffer">Posts</a><a class="menu-item" href="/tags/" title="" rel="noopener noreffer">Tags</a><a class="menu-item" href="/categories/" title="" rel="noopener noreffer">Categories</a><a class="menu-item" href="/about/" title="" rel="noopener noreffer">About</a><a href="javascript:void(0);" class="menu-item theme-switch" title="Switch Theme">
                <i class="fas fa-adjust fa-fw"></i>
            </a></div>
    </div>
</header>

<script>
    window.desktopHeaderMode = "fixed";
    window.mobileHeaderMode = "auto";
</script>
<main class="main">
                <div class="container"><article class="page single"><h1 class="single-title animated flipInX">Genome Analysis</h1><div class="post-meta">
            <div class="post-meta-line"><span class="post-author"><a href="/" title="Author" rel="noopener noreffer author" class="author"><i class="fas fa-user-circle fa-fw"></i>Op1um</a>
</span></div>
            <div class="post-meta-line"><i class="far fa-calendar-alt fa-fw"></i><time datetime=2020-03-26>2020-03-26</time>&nbsp;
                <i class="fas fa-pencil-alt fa-fw"></i>about 737 words&nbsp;
                <i class="far fa-clock fa-fw"></i>4 min&nbsp;</div>
        </div><div class="content" id="content"><p>I have long hesitated to order an analysis of my genome. I found it interesting to learn more about myself through genome. But on the other hand I am against the way in which cheap genome analysis companies try to use such personal data. In this article I explain how I had my genome analyzed, I give some advice to protect your personal data as much as possible, and how I proceeded to manually do a much deeper analysis of the raw genome data.</p>
<p>The scripts described in this article are on my repository: <a href="https://github.com/0p1um/genome-analysis">https://github.com/0p1um/genome-analysis</a></p>
<h2 id="order-a-genome-analysis">Order a genome analysis</h2>
<p>First of all, I think that only residents of the European Union should order these genomes analyzes. Because to my knowledge it is only in the European Union that we have laws that sufficiently protect personal data.</p>
<p>That said, I ordered a genome analysis kit from 23 and me, one of the leading companies in this field. I paid 100 dollars, and I got a kit, it&rsquo;s just a test tube to fill with saliva and send back. A few weeks later, genome analysis is available in the user area of the 23 and me website.</p>
<h3 id="tips-to-protect-protect-your-personal-data">Tips to protect protect your personal data</h3>
<ul>
<li>Refuse that 23 and me keep your saliva sample.</li>
<li>Refuse that your genome be compared with the genome of the other 23 and me clients to find genomic relatives.</li>
<li>Recover all the personal data that 23 and me has on you, including the analysis of your genome in raw format, it is this file that I analyze manually below.</li>
<li>Once you have recovered your data, ask 23 and me to erase all your personal data.</li>
</ul>
<h2 id="raw-genome-analysis">Raw genome analysis</h2>
<h3 id="generalities-about-genome">Generalities about genome</h3>
<p>The genome is information.
The information is stored in 22 chromosomes, a chromosome is a string of characters encoding the information,
The information of the genome is coded in a biological language made up of molecules called nucleotide, there are 4 different nucleotide: A, C, G and T. The nucleotide can unite by complementarity, A with T, and C with G. In chromosomes (before cell division), there are 2 strings of nucleotides parallel to each other, each nucleotide in one of the strings is associated with its complementary nucleotide in the other string. It is like the strings are mirrored.
This is why these 2 nucleotide strings contain the same information.</p>
<p>Genetic information is similar to 99.9% for humans.
SNPs, &ldquo;Single-nucleotide polymorphism&rdquo;, are possible variations of the nucleotide found at specific locations in the genome.</p>
<h3 id="content-of-the-raw-genome-file">Content of the raw genome file</h3>
<p>The raw genome file provided by 23 and me is a simple text file, structured like a table whose columns are separated by tabs.</p>
<p>Example of a line contained in the file:</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-fallback" data-lang="fallback">rs548049170     1       69869   TT
</code></pre></td></tr></table>
</div>
</div><p>Each line corresponds to an SNP, and the elements of the line separated by tabs are:</p>
<ul>
<li>The SNP identifier, this identifier is used in scientific studies.</li>
<li>The SNP chromosome.</li>
<li>The position of the SNP in the chromosome. This position is given in the &ldquo;positive&rdquo; direction (one of the ways genomiic information is read)</li>
<li>The variation of nucleotide that has been found in your genome.</li>
</ul>
<h3 id="analysis-script">Analysis script</h3>
<p>I made the script in python3 to find information about the SNPs in my raw genome file.</p>
<p>Run it with:</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-fallback" data-lang="fallback">./get_snp_data.py &lt;path_to_raw_23andme_genome_file&gt;
</code></pre></td></tr></table>
</div>
</div><p>The script will first get a list of all the SNPs that have a page on the site [snpedia] (<a href="http://snpedia.com/),">http://snpedia.com/),</a> it is a public site that compiles scientific information about SNPs.</p>
<p>Then, the genome file in argument of the script is read, and for each SNP contained in this file, if a page for it exists on [snpedia] (<a href="http://snpedia.com">http://snpedia.com</a>), the page is retrieved, and informations is extracted from it.</p>
<p>Note that if the SNP page uses the &ldquo;minus&rdquo; direction of genome reading, the result is modified to correspond with the &ldquo;positive&rdquo; direction as in the 23 and me genome file.</p>
<p>Finally, all this information is written as a json in the <code>snps_analysis.json</code> file.</p>
<h3 id="script-to-read-the-analysis">Script to read the analysis</h3>
<p>I did another python3 script to read the json file generated previously in the command line.</p>
<p>Run the following command:</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-fallback" data-lang="fallback">./format_snps_analysis.py &lt;path_to_snps_analysis_json&gt;
</code></pre></td></tr></table>
</div>
</div><p>It will display summaries of what your SNPs correspond to, the value of SNPs, their identifiers, and the link [snpedia] (<a href="http://snpedia.com/">http://snpedia.com/</a>) to their pages.</p>
<p>Sample of output from that script:</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-fallback" data-lang="fallback">---------------

lower risk for skin cancer

rs1799782       GG      https://snpedia.org/index.php/Rs1799782(C;C)
---------------
</code></pre></td></tr></table>
</div>
</div></div><div class="post-footer" id="post-footer">
    <div class="post-info">
        <div class="post-info-line">
            <div class="post-info-mod">
                <span>The article was updated on 2020-03-26</span>
            </div>
            <div class="post-info-license"></div>
        </div>
        <div class="post-info-line">
            <div class="post-info-md"><span>
                            <a class="link-to-markdown" href="/genome-analysis/index.md" target="_blank">Read Markdown</a>
                        </span></div>
            <div class="post-info-share">
                <span><a href="javascript:void(0);" title="Share on Twitter" data-sharer="twitter" data-url="https://op1um.blog/genome-analysis/" data-title="Genome Analysis" data-via="Op1um1" data-hashtags="open-source,genome,dna,python"><i class="fab fa-twitter fa-fw"></i></a><a href="javascript:void(0);" title="Share on Linkedin" data-sharer="linkedin" data-url="https://op1um.blog/genome-analysis/"><i class="fab fa-linkedin fa-fw"></i></a><a href="javascript:void(0);" title="Share on Hacker News" data-sharer="hackernews" data-url="https://op1um.blog/genome-analysis/" data-title="Genome Analysis"><i class="fab fa-hacker-news fa-fw"></i></a><a href="javascript:void(0);" title="Share on Reddit" data-sharer="reddit" data-url="https://op1um.blog/genome-analysis/"><i class="fab fa-reddit fa-fw"></i></a></span>
            </div>
        </div>
    </div>

    <div class="post-info-more">
        <section><i class="fas fa-tags fa-fw"></i>&nbsp;<a href="/tags/open-source">open-source</a>,&nbsp;<a href="/tags/genome">genome</a>,&nbsp;<a href="/tags/dna">dna</a>,&nbsp;<a href="/tags/python">python</a></section>
        <section>
            <span><a href="javascript:window.history.back();">Back</a></span>&nbsp;|&nbsp;<span><a href="/">Home</a></span>
        </section>
    </div>

    <div class="post-nav">
            <a href="/osint-find-people-using-tinder/" class="next" rel="next" title="Osint Find People Using Tinder">Osint Find People Using Tinder<i class="fas fa-angle-right fa-fw"></i></a></div>
</div>
<div class="comment"><div id="gitalk"></div><script>
            document.addEventListener("DOMContentLoaded", function(event) {
                var gitalk = new Gitalk({
                    id: '2020-03-26 13:23:09 \x2b0100 CET',
                    title: 'Genome Analysis',
                    clientID: '98a7c547a7f01529e488',
                    clientSecret: '97ba580896f2689c40a6cb7f8d05f5ef96012257',
                    repo: '0p1um.github.io',
                    owner: '0p1um',
                    admin: ['0p1um'],
                    body: decodeURI(location.href),
                });
                gitalk.render('gitalk');
            });
        </script>
        <noscript>
            Please enable JavaScript to view the <a href="https://github.com/gitalk/gitalk">comments powered by Gitalk.</a>
        </noscript></div>
    </article></div>
            </main><footer class="footer">
    <div class="copyright"><div class="copyright-line">Powered by <a href="https://gohugo.io/" target="_blank" rel="external nofollow noopener noreffer">Hugo</a> | Theme - <a href="https://github.com/dillonzq/LoveIt" target="_blank" rel="external nofollow noopener noreffer"><i class="far fa-heart fa-fw"></i> LoveIt</a>
        </div>

        <div class="copyright-line"><i class="far fa-copyright fa-fw"></i><span itemprop="copyrightYear">2019 - 2020</span><span class="author" itemprop="copyrightHolder">&nbsp;<a href="/" target="_blank">Op1um</a></span></div>
    </div>
</footer>
</div><a href="#" class="dynamic-to-top animated faster" id="dynamic-to-top">
            <i class="fas fa-chevron-up fa-fw"></i>
        </a><script>
        document.addEventListener('DOMContentLoaded', function () {
            lightGallery(document.getElementById('content'), {
                selector: '.lightgallery',
                speed: 400,
                hideBarsDelay: 2000,
                thumbnail: true,
                exThumbImage: 'data-thumbnail',
                thumbWidth: 80,
                thumbContHeight: 80,
            });
        });
    </script><script>
        document.addEventListener('DOMContentLoaded', function () {
            renderMathInElement(document.body, {
                delimiters: [
                    { left: '$$', right: '$$', display: true },
                    { left: '\\[', right: '\\]', display: true },{ left: '$', right: '$', display: false },
                    { left: '\\(', right: '\\)', display: false },],
                strict: false,
            });
        });
    </script><link rel="stylesheet" href="/lib/gitalk/gitalk.min.css"><link rel="stylesheet" href="/lib/lightgallery/lightgallery.min.css"><link rel="stylesheet" href="/lib/katex/katex.min.css"><link rel="stylesheet" href="/lib/katex/copy-tex.min.css"><script src="/lib/gitalk/gitalk.min.js"></script><script src="/lib/smooth-scroll/smooth-scroll.polyfills.min.js"></script><script src="/lib/sharer/sharer.min.js"></script><script src="/lib/lazysizes/lazysizes.min.js"></script><script src="/lib/lightgallery/lightgallery.min.js"></script><script src="/lib/lightgallery/lg-thumbnail.min.js"></script><script src="/lib/lightgallery/lg-zoom.min.js"></script><script src="/lib/katex/katex.min.js"></script><script src="/lib/katex/auto-render.min.js"></script><script src="/lib/katex/copy-tex.min.js"></script><script src="/lib/katex/mhchem.min.js"></script><script src="/js/theme.min.js"></script></body>
</html>
